msm/src/serialization/constraints.rs

```diff
@@ -15,13 +16,17 @@ pub struct Env<Fp> {
     /// folding for instance.
     pub constraints: Vec<(usize, Expr<ConstantExpr<Fp>, Column>)>,
     pub constrain_index: usize,
+    pub rangecheck4_lookups: Vec<Lookup<E<Fp>>>,
```

Comment:
I wonder if it is really a good idea to have the lookups split into different variable names. Looks more general to have just a single lookups vector, and then all of them get pushed into it.

Reply:
Yeah, probably we should think of a way to generalize them to just have one. And then if you really need the distinction of different tables, think of a different way to split them but "inside" the `lookup` type/vector/struct/whatever. Maybe for a follow up.

Reply:
Yes, agreed. This is ugly atm, and we must generalize. Let's do it quickly after everything works.